docker-images/kafka-based/kafka/scripts/kafka_run.sh

```diff
@@ -63,8 +63,9 @@ if [ "$STRIMZI_KRAFT_ENABLED" = "true" ]; then
     echo "Kraft storage is already formatted"
   fi
 
-  touch /var/opt/kafka/kafka-ready
-  touch /var/opt/kafka/zk-connected
+  rm -f /var/opt/kafka/kafka-ready 2> /dev/null
+  KAFKA_OPTS="${KAFKA_OPTS} -javaagent:$(ls "$KAFKA_HOME"/libs/kafka-agent*.jar)=/var/opt/kafka/kafka-ready"
+  export KAFKA_OPTS
```

Comment:
If we want to keep this code as is, I'm fine and can live with it. But I wonder if it would be nicer to have a flag in the readiness probe to indicate KRaft (or read it from the environment variable) and configure the probe just once. But as I said, if you prefer this method, I'm fine with it.